game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -749,27 +768,67 @@ public JComponent getUiComponent() {
 
       @Override
       public void save(final SaveContext context) {
-        final String username = usernameField.getText();
-        context.setValue(usernameSetting, username.isEmpty() ? null : username.toCharArray());
+        // Only save when value changed
+        if (usernameField
+            .getText()
+            .equals(usernameSetting.getValue().map(String::new).orElse(""))) {
+          return;
+        }
+        try {
+          BackgroundTaskRunner.runInBackground(
```

Comment:
Side-note, we really need to update BackTasRunner to be closeable. If anything crashes  while this is running we can run into problems. Notably if any exception is uncaught, then is handled and shows the error dialog, the background task will not complete until the dialog is closed and the dialog cannot be closed because the background task runner is modal and does not give up UI focus.

We'll probably want to test this out and/or fix the background task runner. We could see problems if the lobby were not responding or is being restarted, in such a case we'd want a graceful failure.